java-util/src/main/java/io/druid/java/util/common/parsers/ParserUtils.java

```diff
@@ -92,6 +93,17 @@ public static String stripQuotes(String input)
     return input;
   }
 
+  public static boolean isValidTimeZone(String timeZone)
```

Comment:
Suggested to prepare a static map from `String` to joda-time's `DateTimeZone`, using `TimeZone.getAvailableIDs()` as keys. Then add a method `tryConvertToTimeZone(string)`, that is implemented just as `map.get()`, i. e. it returns `null` if time zone appears to not be valid. And then use this method in `TimestampParser`.